ludwig/trainers/trainer.py

```diff
@@ -540,23 +555,36 @@ def run_evaluation(
                 self.increase_batch_size_eval_split,
                 early_stopping_steps,
                 self.skip_save_model,
+                checkpoint_manager,
             )
         else:
             # There's no validation, so we save the model.
-            if self.is_coordinator() and not self.skip_save_model:
-                self.model.save(save_path)
+            if not self.skip_save_model:
+                logger.info("Saving model.\n")
+                checkpoint_manager.save_best(progress_tracker.steps)
+                self.callback(lambda c: c.on_save_best_checkpoint(self, progress_tracker, save_path))
```

Comment:
Does this save on the coordinator automatically?

Reply:
Ohhhh I see, they're all CoordinatorCheckpoint classes that internally do the `is_coordinator()` checks, so this makes sense!

Reply:
Good point, though, for hyperopt we only want to save the model on the coordinator. Updated.